engine/src/engine_api.rs

```diff
@@ -13,7 +13,6 @@ pub enum WriteOperation<'a> {
 }
```

Comment:
Maybe it's more conventional to use nested struct, e.g. 
```suggestion
pub enum WriteOperation<'a> {
    /// `Put` operation
    Put {
        /// The table name
        pub(crate) table: &'a str,
        /// Key
        pub(crate) key: Vec<u8>,
        /// Value
        pub(crate) value: Vec<u8>,
    },
    // ...
```

And put the construction functions to `WriteOperation`, eg. `WriteOperation::new_put(...)` `WriteOperation::new_delete()`.

Reply:
Fixed